pennylane/optimize/rotosolve.py

```diff
@@ -97,8 +141,33 @@ class RotosolveOptimizer:
     separately reconstructing the cost function with respect to each circuit parameter,
     while keeping all other parameters fixed.
 
+    Args:
+        optimizer (str or callable): Optimizer to use for the substeps of Rotosolve
```

Comment:
I have a minor worry that calling this argument `optimizer` might be confusing to users who might think 'I just want to use Rotosolve as my optimizer'.

Reply:
So two thoughts here:

- Can we set a default?
- Is there a modification to the argument name? Maybe, um, `suboptimizer`? lol

Reply:
- There is a default down the pipeline in `_prepare_optimizer`. Should it be moved? (Relates to the discussion above).
- `substep_optimizer` and `substep_optimizer_kwargs` or `substep_kwargs`?

Reply:
> There is a default down the pipeline in _prepare_optimizer. Should it be moved? (Relates to the discussion above).

I think so yes, the way it currently looks in the user-facing docstring in the documentation, it looks like you _have_ to pass an optimizer option here. It would be clearer to have `optimizer="brute"` at this level.

> `substep_optimizer` and `substep_optimizer_kwargs` or `substep_kwargs`?

my preference is always shorter, so `substep_kwargs`?